gr-filter/lib/rational_resampler_base_impl.cc

```diff
@@ -0,0 +1,188 @@
+/* -*- c++ -*- */
+/*
+ * Copyright 2004,2010,2012,2018 Free Software Foundation, Inc.
+ *
+ * This file is part of GNU Radio
+ *
+ * GNU Radio is free software; you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation; either version 3, or (at your option)
+ * any later version.
+ *
+ * GNU Radio is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with GNU Radio; see the file COPYING.  If not, write to
+ * the Free Software Foundation, Inc., 51 Franklin Street,
+ * Boston, MA 02110-1301, USA.
+ */
+
+#ifdef HAVE_CONFIG_H
+#include "config.h"
+#endif
+
+#include "rational_resampler_base_impl.h"
+#include <gnuradio/io_signature.h>
+#include <volk/volk.h>
+#include <stdexcept>
+
+namespace gr {
+  namespace filter {
+
+    template <class IN_T, class OUT_T, class TAP_T>
+    typename rational_resampler_base<IN_T,OUT_T,TAP_T>::sptr
+    rational_resampler_base<IN_T,OUT_T,TAP_T>::make(unsigned interpolation,
+		      unsigned decimation,
+		      const std::vector<TAP_T> &taps)
+    {
+      return gnuradio::get_initial_sptr
+	(new rational_resampler_base_impl<IN_T,OUT_T,TAP_T>(interpolation, decimation, taps));
+    }
+
+    template <class IN_T, class OUT_T, class TAP_T>
+    rational_resampler_base_impl<IN_T,OUT_T,TAP_T>::rational_resampler_base_impl(unsigned interpolation,
+			     unsigned decimation,
+			     const std::vector<TAP_T> &taps)
+    : block("rational_resampler_base<IN_T,OUT_T,TAP_T>",
+	       io_signature::make(1, 1, sizeof(IN_T)),
+	       io_signature::make(1, 1, sizeof(OUT_T))),
+      d_history(1),
+      d_interpolation(interpolation),
+      d_decimation(decimation),
+      d_ctr(0),
+      d_firs(interpolation),
+      d_updated(false)
+    {
+      if(interpolation == 0)
+	throw std::out_of_range("rational_resampler_base_impl<IN_T,OUT_T,TAP_T>: interpolation must be > 0");
```

Comment:
indents borked? My compiler complains about such things.

Reply:
idents are borked throughout a lot of files and are mostly unfixed.

Reply:
:/ ok, let's leave it like that